### DISCUSSION TITLE
s3 bucket SSL connection

### DISCUSSION THREAD

**Discussion by SchmidtJonathan1** [argoproj/argo-workflows#6111] [General]
Hello,
I've some problems with my SSL connection to the s3 bucket. The connection was already checked without SSL, it works perfect.
The line "insecure: false" and my secret are added to the template YAML data.
After a 40 second run, it write the following output:

level=info msg="Getting from s3 (endpoint: [NameOfS3Bucket], bucket: ai-library-01, key: flake_analysis/training/zips/flakes.tar.gz) to /argo/inputs/artifacts/flakestraindata.tmp"
level=warning msg="Failed get file: Get https://[NameOfS3Bucket]/ai-library-01/?location=: **x509: certificate signed by unknown authority**"
level=info msg="S3 Load path: /argo/inputs/artifacts/flakestraindata.tmp, key: flake_analysis/training/zips/flakes.tar.gz"
level=info msg="Creating minio client [NameOfS3Bucket] using static credentials"

Is there a possibility to run it with a self signed SSL certificate?

Thanks for your help

**Comment by SchmidtJonathan1** [comment]
Hello,
pls close this conversation. Thanks

**Reply by rush2sk8** [reply]
How did you fix this?

**Reply by SchmidtJonathan1** [reply]
Hello, I didn't... Sorry. 
Decided to do it without ssl connection.

**Reply by rush2sk8** [reply]
I found the answer. You have to rebuild the argoexec container and manually trust the ca cert. I did it in windows workflows and then used certoc.exe to trust the ca certs.

**Reply by SchmidtJonathan1** [reply]
Thank you

**Comment by SchmidtJonathan1** [comment]
Close its now three years old
